test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/PrimitiveTypesTests/Tests/DurationTests.cs

```diff
@@ -0,0 +1,265 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DurationTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.ObjectModel;
+using System.Xml;
+using Microsoft.AspNetCore.OData;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.TestCommon.Common;
+using Microsoft.OData.Client.E2E.TestCommon.Helpers;
+using Microsoft.OData.Client.E2E.Tests.Common.Client.Default.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.Default;
+using Microsoft.OData.Client.E2E.Tests.PrimitiveTypesTests.Server;
+using Microsoft.OData.Edm;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.PrimitiveTypesTests.Tests;
+
+public class DurationTests : EndToEndTestBase<DurationTests.TestsStartup>
+{
+    private readonly Uri _baseUri;
+    private readonly Container _context;
+    private readonly IEdmModel _model;
+
+    public class TestsStartup : TestStartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.ConfigureControllers(typeof(PrimitiveTypesTestsController), typeof(MetadataController));
+
+            services.AddControllers().AddOData(opt =>
+                opt.EnableQueryFeatures().AddRouteComponents("odata", DefaultEdmModel.GetEdmModel()));
+        }
+    }
+
+    public DurationTests(TestWebApplicationFactory<TestsStartup> fixture) 
+        : base(fixture)
+    {
+        if(Client.BaseAddress == null)
+        {
+            throw new ArgumentNullException(nameof(Client.BaseAddress), "Base address cannot be null");
+        }
+
+        _baseUri = new Uri(Client.BaseAddress, "odata/");
+
+        _context = new Container(_baseUri)
+        {
+            HttpClientFactory = HttpClientFactory
+        };
+
+        _model = DefaultEdmModel.GetEdmModel();
+        ResetDefaultDataSource();
+    }
+
+    public static IEnumerable<object[]> MimeTypesData
+    {
+        get
+        {
+            yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata };
+            yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterMinimalMetadata };
+            yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterNoMetadata };
+        }
+    }
+
+    #region Entity Set
+
+    [Theory]
+    [MemberData(nameof(MimeTypesData))]

```

Comment:
If these mime types are constants, then I'd suggest using inline data as so:

```suggestion
    [InlineData(MimeTypes.ApplicationJson + ODataParameterFullMetadata)]
    [InlineData(MimeTypes.ApplicationJson + MimeTypes.ODataParameterMinimalMetadata)]
    [InlineData(MimeTypes.ApplicationJson + MimeTypes.ODataParameterNoMetadata)]
```

Why? This makes the tests easier to read. It's easier to know what is being tested without having to search for the definition of the member data collection. We should strive to keep the tests self-contained when feasible, so that we don't have to look at other source code to understand what the test is trying to achieve.

If the fact that these mime types are constant is not important to the feature, we could even go further and write them out as string literals.

Reply:
I have created them as constants and passed them using InlineData as recommended